src/MongoDB/Monitoring/TopologyChangedEvent.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2021-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <php.h>
+#include <Zend/zend_interfaces.h>
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "phongo_compat.h"
+#include "php_phongo.h"
+
+zend_class_entry* php_phongo_topologychangedevent_ce;
+
+/* {{{ proto MongoDB\Driver\Monitoring\TopologyChangedEvent TopologyChangedEvent::getNewDescription()
+   Returns this event's new description */
+PHP_METHOD(TopologyChangedEvent, getNewDescription)
+{
+	mongoc_topology_description_t*     topology_description;
+	php_phongo_topologychangedevent_t* intern = Z_TOPOLOGYCHANGEDEVENT_OBJ_P(getThis());
+
+	PHONGO_PARSE_PARAMETERS_NONE();
+
+	topology_description = mongoc_topology_description_new_copy(intern->new_topology_description);
+
+	phongo_topologydescription_init(return_value, topology_description);
+} /* }}} */
+
+/* {{{ proto MongoDB\Driver\Monitoring\TopologyChangedEvent TopologyChangedEvent::getPreviousDescription()
+   Returns this event's previous description */
+PHP_METHOD(TopologyChangedEvent, getPreviousDescription)
+{
+	mongoc_topology_description_t*     topology_description;
+	php_phongo_topologychangedevent_t* intern = Z_TOPOLOGYCHANGEDEVENT_OBJ_P(getThis());
+
+	PHONGO_PARSE_PARAMETERS_NONE();
+
+	topology_description = mongoc_topology_description_new_copy(intern->old_topology_description);
+
+	phongo_topologydescription_init(return_value, topology_description);
+} /* }}} */
+
+/* {{{ proto string TopologyChangedEvent::getTopologyId()
+   Returns this event's topology id */
+PHP_METHOD(TopologyChangedEvent, getTopologyId)
+{
+	php_phongo_topologychangedevent_t* intern = Z_TOPOLOGYCHANGEDEVENT_OBJ_P(getThis());
+	char                               topology_id[25];
+
+	PHONGO_PARSE_PARAMETERS_NONE();
+
+	bson_oid_to_string(&intern->topology_id, topology_id);
+	RETVAL_STRING(topology_id);
+} /* }}} */
+
+/**
+ * Event thrown when the driver observes a change in any of the servers
+ * it is connected to or a change in the overall server topology.
```

Comment:
What documentation inspired this comment? It doesn't agree with the definition of a TopologyChangedEvent in the [SDAM Monitoring spec](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring-monitoring.rst#events).

Reply:
I got the description from the [libmongoc docs](http://mongoc.org/libmongoc/current/mongoc_apm_topology_changed_t.html) - which should we follow?

Reply:
I'm not aware of any other class files that have a comment like this documenting the class itself. I'd just suggest we remove this. If anything, this is the kind of documentation that we'd ultimately want in the PHP.net API documentation.

Reply:
Ok - I got it from the other event classes like `CommandStartedEvent` which contain a similar comment https://github.com/mongodb/mongo-php-driver/blob/6a91e0c8cfc23f5fb0ba90ea125856eaa30ea9db/src/MongoDB/Monitoring/CommandStartedEvent.c#L156

Reply:
Noted. I still think we can do without it here.